src/clandinin_lab_to_nwb/brezovec/brezovecimaginginterface.py

```diff
@@ -1,21 +1,119 @@
     """
-    Data Interface for writing the Clandinin lab imaging data to NWB file using BrezovecMultiPlaneImagingExtractor.
+    Data Interface for writing the functional imaging data (GCaMP6f) for Clandinin lab to NWB file using BrezovecMultiPlaneImagingExtractor.
     """
 
     Extractor = BrezovecMultiPlaneImagingExtractor
 
     def __init__(self, folder_path: FolderPathType, stream_name: str, verbose: bool = True):
+        """
+        Initialize reading of NIfTI files.
 
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            The path to the folder that contains the NIfTI image files (.nii) and configuration files (.xml) from Bruker system.
+        stream_name: str, optional
+            The name of the recording channel.
+        verbose : bool, default: True
+        """
+        super().__init__(
+            folder_path=folder_path,
+            stream_name=stream_name,
+            verbose=verbose,
+        )
+
+    @classmethod
+    def get_streams(cls, folder_path) -> dict:
+        streams = BrezovecMultiPlaneImagingExtractor.get_streams(folder_path=folder_path)
+        return streams
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+
+        xml_metadata = self.imaging_extractor.xml_metadata
+        session_start_time = parse(xml_metadata["date"])
+        metadata["NWBFile"].update(session_start_time=session_start_time)
+
+        # channel_name = 'Green'
+        # metadata["Ophys"]["OpticalChannel"][0].update(
+        #     name=channel_name,
+        #     emission_lambda=513.0,
+        #     description="Green channel of the microscope, 525/50 nm filter.",
+        # )
+
+        device_name = "BrukerFluorescenceMicroscope"
+        description = f"Version {xml_metadata['version']}"
+        metadata["Ophys"]["Device"][0].update(
+            name=device_name,
+            description=description,
+        )
+
+        imaging_plane_name = "ImagingPlaneGCaMP6fFunctional"
+        imaging_plane_metadata = metadata["Ophys"]["ImagingPlane"][
+            0
+        ]  # should it be photon_series_index pr imaging_plane_index
+        imaging_plane_metadata.update(
+            name=imaging_plane_name,
+            # optical_channel=channel_name,
+            device=device_name,
+            excitation_lambda=920.0,
+            indicator="GCaMP6f",
+            imaging_rate=self.imaging_extractor.get_sampling_frequency(),
+        )
+
+        two_photon_series_metadata = metadata["Ophys"]["TwoPhotonSeries"][0]  # should it be photon_series_index
+        two_photon_series_metadata.update(
+            name="TwoPhotonSeriesFunctionalGreen",
+            imaging_plane=imaging_plane_name,
+            format=".nii",
+            scan_line_rate=1 / float(xml_metadata["scanLinePeriod"]),  # why I cannot see it in the NWBFile?
+            rate=self.imaging_extractor.get_sampling_frequency(),
```

Comment:
Both scan_line_rate and rate are not shown in the NWB file generated.

Reply:
I think it might be a problem with your hdf5 viewer?

They show for me:

```python
# Load the file with pynwb
from pynwb import NWBHDF5IO
io = NWBHDF5IO(str(file_path), mode='r')
nwbfile = io.read()
nwbfile
nwbfile.acquisition["TwoPhotonSeriesFunctionalGreen"]
```

Output:
```
TwoPhotonSeriesFunctionalGreen pynwb.ophys.TwoPhotonSeries at 0x140482468723248
Fields:
  comments: no comments
  conversion: 1.0
  data: <HDF5 dataset "data": shape (10, 256, 128, 49), type "<u2">
  description: Imaging data acquired (GCaMP6f) from the Bruker Two-Photon Microscope and transform to NIfTI. Used for functional imaging readout
  dimension: <HDF5 dataset "dimension": shape (3,), type "<i8">
  field_of_view: <HDF5 dataset "field_of_view": shape (3,), type "<f8">
  format: .nii
  imaging_plane: GCaMP6f_functional pynwb.ophys.ImagingPlane at 0x140482468723536
Fields:
  conversion: 1.0
  description: The plane or volume being imaged by the microscope.
  device: BrukerFluorescenceMicroscope pynwb.device.Device at 0x140482468721568
Fields:
  description: Version 5.5.64.100

  excitation_lambda: 920.0
  imaging_rate: 1.8797144377675852
  indicator: GCaMP6f
  location: unknown
  optical_channel: (
    Green <class 'pynwb.ophys.OpticalChannel'>
  )
  unit: meters

  offset: 0.0
  rate: 1.8797144377675852
  resolution: 2.61117299284587e-06
  scan_line_rate: 15833.056254848874
  starting_time: 0.0
  starting_time_unit: seconds
  unit: px
  ```